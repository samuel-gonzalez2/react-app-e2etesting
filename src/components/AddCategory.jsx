import { useState } from "react";

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("one punch");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    console.log(inputValue);
    }

  return (
    <form onSubmit={event => onSubmit(event)}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        onChange={handleInputChange}
      />
    </form>
  );
};
