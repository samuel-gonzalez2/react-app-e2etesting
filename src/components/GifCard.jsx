import PropTypes from 'prop-types';
import React from 'react'

export const GifCard = ({ title, url }) => {
  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}


GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}