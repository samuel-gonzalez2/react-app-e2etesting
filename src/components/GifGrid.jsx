import { GifCard } from "./GifCard";
import PropTypes from "prop-types";

//import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {



  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <p>Cargando</p>}

      <div className="card-grid">
        {images.map(img => (
            <GifCard key={img.id} {...img}/>
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};