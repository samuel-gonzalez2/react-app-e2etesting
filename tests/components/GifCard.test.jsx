import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard";

describe('Pruebas en GifCard Component', () => {

    const title = 'Simpsons';
    const url = 'https://media1.giphy.com/media/l0Hl6qQ7yvZ7O/200w.gif';

    test('Debe de hacer match con el snapshot', () => {
        const {container } = render(<GifCard title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    } )

    test('Debe mostrar la imagen con el URL y ALT indicado', () => {
        render ( <GifCard title={title} url={url}/> )

        const { src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    });

    test('Debe mostrar el tÃ­tulo con el texto indicado', () => {
        render ( <GifCard title={title} url={url}/> )

        const { textContent } = screen.getByRole('img')
        expect(screen.getByText(title)).toBeTruthy()
        
    }
    )

});