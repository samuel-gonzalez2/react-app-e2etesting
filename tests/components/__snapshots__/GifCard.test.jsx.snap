// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifCard Component Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Simpsons"
      src="https://media1.giphy.com/media/l0Hl6qQ7yvZ7O/200w.gif"
    />
    <p>
      Simpsons
    </p>
  </div>
</div>
`;
